lib/enhanceReport.js

```diff
@@ -102,22 +125,48 @@ function createScreenshotsContextList(screenshots, titleSuffix, mochawesomeOptio
     value:
       mochawesomeOptions.embeddedScreenshots === true
         ? convertImageToBase64(screenshotsDir, screenshot.value)
-        : encodeScreenshotPath(`screenshots${screenshot.value}`),
+        : encodeMediaPath(`screenshots${screenshot.value}`),
   }));
 }
 
+function createVideoContext(video, mochawesomeOptions, baseFolder) {
+  video = video.replace(baseFolder, '').concat('.mp4')
+  return {
+    title: 'Spec video recording',
+    value: encodeMediaPath(`videos/${video}`)
+  };
+}
+
 function convertImageToBase64(screenshotsDir, imagePath) {
   const imgPath = path.join(screenshotsDir, imagePath);
   const contents = 'data:image/png;base64, ' + fse.readFileSync(imgPath, { encoding: 'base64' });
   return contents;
 }
 
-function encodeScreenshotPath(p) {
+function encodeMediaPath(p) {
   const { dir, name, ext } = path.parse(p);
 
   return `${dir}/${encodeURIComponent(name)}${ext}`;
 }
 
+function extrapolateBaseFolder(results) {
+  //Snatched from https://rosettacode.org/wiki/Find_common_directory_path#JavaScript
+  const splitStrings = (a, sep = '\\') => a.map(i => i.split(sep));
+  const elAt = i => a => a[i];
+  const rotate = a => a[0].map((e, i) => a.map(elAt(i)));
+  const allElementsEqual = arr => arr.every(e => e === arr[0]);
+  const commonPath = (input, sep = '\\') => rotate(splitStrings(input, sep))
+      .filter(allElementsEqual).map(elAt(0)).join(sep);
```

Comment:
@LironEr Thinking about it, you file path issue might be because I coded it on a windows machine and the folder separator is probably not the same. Replacing `'\\'` with `path.sep` should fix the issue

Reply:
Awesome, works great!